codegen/src/test/resources/software/amazon/awssdk/codegen/poet/auth/scheme/query-auth-scheme-interceptor.java

```diff
@@ -100,6 +101,14 @@ private <T extends Identity> SelectedAuthScheme<T> trySelectAuthScheme(AuthSchem
                 .add(() -> String.format("'%s' does not have an identity provider configured.", authOption.schemeId()));
             return null;
         }
+        HttpSigner<T> signer;
+        try {
+            signer = authScheme.signer();
+        } catch (RuntimeException e) {
```

Comment:
Why are we not catching specific exception like ClassNotFoundException ?

Reply:
- I wanted it to be more generic to catch other issues in the future. I don't think we should only be looking for classnotfound
- The V4a authscheme doesn't throw classnotfound in this case

Reply:
- What exception is thrown in this case?
- What is the actual low-level exception sent in cases when CRT is not in the classpath? 
- Is it possible that the low-level exception is suppressed in the expression and can be retrieved using getCause?
- Why do you think we should catch generic runtime exceptions?"

Reply:
- It throws a `RuntimeException`: https://github.com/aws/aws-sdk-java-v2/blob/b0fd0e65928363573bea404bf1498713c032d538/core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/util/OptionalDependencyLoaderUtil.java#L50

In any case, we're not solving for *just* CRT. This fix is intended to be more broad so that we catch any issues retrieving the signer from the auth scheme. By catching `RuntimeException` we ensure that we can handle a broader set of issues that can happen when trying to retrieve the signer. In all case, we should be moving on to the next available auth scheme.